source/utilities.tex

```diff
@@ -19492,3 +19493,1887 @@
+\pnum
+The \fmtgrammarterm{format-spec} field contains
+\defnx{format specifications}{format specification!format string}
+that define how the value should be presented,
+including such details as
+field width, alignment, padding, and decimal precision.
+Each type can define its own
+formatting mini-language or
+interpretation of the \fmtgrammarterm{format-spec} field.
+The syntax of format specifications is as follows:
+
+% FIXME: Listing chrono-format-spec here is a mistake. We should have a clean
+% separation between the extensibility mechanism and the extensions to it.
+\begin{ncbnf}
+\fmtnontermdef{format-spec}\br
+    \terminal{:} std-format-spec\br
+    \terminal{:} chrono-format-spec\br
+    \terminal{:} custom-format-spec
+\end{ncbnf}
+
+\begin{ncbnf}
+\fmtnontermdef{custom-format-spec}\br
+    \textnormal{user-defined syntax dependent on the type of the argument}
+\end{ncbnf}
+
+\begin{ncbnf}
+\fmtnontermdef{std-format-spec}\br
+    \opt{fill-and-align} \opt{sign} \opt{\terminal{\#}} \opt{\terminal{0}} \opt{width} \opt{precision} \opt{type}
+\end{ncbnf}
+
+\begin{ncbnf}
+\fmtnontermdef{fill-and-align}\br
+    \opt{fill} align
+\end{ncbnf}
+
+\begin{ncbnf}
+\fmtnontermdef{fill}\br
+    \textnormal{any character other than \tcode{\{} or \tcode{\}}}
+\end{ncbnf}
+
+\begin{ncbnf}
+\fmtnontermdef{align} \textnormal{one of}\br
+    \terminal{< > \caret}
+\end{ncbnf}
+
+\begin{ncbnf}
+\fmtnontermdef{sign} \textnormal{one of}\br
+    \terminal{+ -} \textnormal{space}
+\end{ncbnf}
+
+\begin{ncbnf}
+\fmtnontermdef{width}\br
+    positive-integer\br
+    \terminal{\{} arg-id \terminal{\}}
+\end{ncbnf}
+
+\begin{ncbnf}
+\fmtnontermdef{precision}\br
+    \terminal{.} nonnegative-integer\br
+    \terminal{.} \terminal{\{} arg-id \terminal{\}}
+\end{ncbnf}
+
+\begin{ncbnf}
+\fmtnontermdef{type} \textnormal{one of}\br
+    \terminal{a A b B c d e E f F g G n o p s x X}
+\end{ncbnf}
+
+where
+\fmtgrammarterm{std-format-spec}
+defines a common formatting mini-language
+supported by fundamental and string types,
+\fmtgrammarterm{chrono-format-spec}
+defines a mini-language for chrono types\iref{time.format}, and
+\fmtgrammarterm{custom-format-spec}
+is a placeholder for user-defined mini-languages.
+\begin{note}
+Some of the formatting options
+are only supported for arithmetic types.
+\end{note}
+
+\pnum
+\begin{note}
+The \fmtgrammarterm{fill} character can be any character
+other than \tcode{\{} or \tcode{\}}.
+The presence of a fill character is signaled by the
+character following it, which must be one of the alignment options.
+If the second character of \fmtgrammarterm{std-format-spec}
+is not a valid alignment option,
+then it is assumed that both the fill character and the alignment option are
+absent.
+\end{note}
+
+\pnum
+The meaning of the various alignment options is as specified in \tref{format.align}.
+\begin{example}
+\begin{codeblock}
+char c = 120;
+string s0 = format("{:6}", 42);         // value of \tcode{s0} is \tcode{"\ \ \ \ 42"}
```

Comment:
There's no normative wording that makes `charT{' '}` or any other character the default fill character.

Reply:
Already noted and filed as an LWG issue. But this is not editorial.